src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs

```diff
@@ -464,7 +472,7 @@ protected override CustomGetCapabilitiesResult Handle(IPowerFxScope scope, Custo
             {
                 var result = new CustomGetCapabilitiesResult();
 
-                var nl = _parent.NL2FxImplementation;
+                var nl = _parent._nlHandlerFactory?.GetNLHandler(scope) ?? _parent.NL2FxImplementation;
```

Comment:
> _parent._nlHandlerFactory?.GetNLHandler(scope) ?? _parent.NL2FxImplementation; [](http://example.com/codeflow?start=24&length=79)

nit - can we put this behind a helper?

     var nl = _parent.GetNLHandler(scope);